rs/nervous_system/integration_tests/src/nervous_system_agent_helpers.rs

```diff
@@ -0,0 +1,538 @@
+                        // more attempts to get the proposal info to find out if the proposal
+                        // actually got executed.
+                        if agent.is_canister_stopped_error(&user_error) {
+                            continue;
+                        } else {
+                            return Err(format!("Error getting proposal info: {:#?}", user_error));
+                        }
+                    }
+                };
+
+                if proposal_info.executed_timestamp_seconds > 0 {
+                    return Ok(proposal_info);
+                }
+                assert_eq!(
+                    proposal_info.failure_reason,
+                    None,
+                    "Execution failed for {:?} proposal '{}': {:#?}",
+                    Topic::try_from(proposal_info.topic).unwrap(),
+                    proposal_info
+                        .proposal
+                        .unwrap()
+                        .title
+                        .unwrap_or("<no-title>".to_string()),
+                    proposal_info.failure_reason
+                );
+                last_proposal_info = Some(proposal_info);
+            }
+            Err(format!(
+                "Looks like proposal {:?} is never going to be executed: {:#?}",
+                proposal_id, last_proposal_info,
+            ))
+        }
+
+        pub async fn nns_get_proposal_info<C: CallCanisters>(
+            agent: &C,
+            proposal_id: ProposalId,
+        ) -> Result<ProposalInfo, C::Error> {
+            nns_agent::governance::get_proposal_info(agent, proposal_id)
+                .await
+                .map(|result| result.unwrap())
+        }
+    }
+
+    pub mod ledger {
+        use super::*;
+        use ic_ledger_core::Tokens;
+        use icp_ledger::{AccountIdentifier, BinaryAccountBalanceArgs, TransferArgs};
+        use icrc_ledger_types::icrc1::transfer::TransferArg;
+
+        pub async fn icrc1_transfer<C: CallCanisters>(
+            agent: &C,
+            transfer_arg: TransferArg,
+        ) -> Result<Nat, icrc_ledger_types::icrc1::transfer::TransferError> {
+            nns_agent::ledger::icrc1_transfer(agent, transfer_arg)
+                .await
+                .unwrap()
+        }
+
+        pub async fn account_balance<C: CallCanisters>(
+            agent: &C,
+            account: &AccountIdentifier,
+        ) -> Tokens {
+            nns_agent::ledger::account_balance(
+                agent,
+                BinaryAccountBalanceArgs {
+                    account: account.to_address(),
+                },
+            )
+            .await
+            .unwrap()
+        }
+
+        pub async fn transfer<C: CallCanisters>(
+            agent: &C,
+            transfer_args: TransferArgs,
+        ) -> Result<u64, icp_ledger::TransferError> {
+            nns_agent::ledger::transfer(agent, transfer_args)
```

Comment:
Is the main point of these helpers to have a shorter, unsafe version of the corresponding functions from nns_agent?

Reply:
I think the only reason they exist is because I performed refactoring in the following way:
1) Implement canister interaction in `ic-nervous-system-agent`.
2) Abstract out helpers from `pocket_ic_helpers.rs` to use `CallCanisters` and move them here.
During these steps, I tried to persist backward compatibility with the existing tests in this crate), as a result
some of these helpers indeed look redundant